### ISSUE TITLE
why there is a tT_loss

### ISSUE THREAD

**Issue by BaohaoLiao** [madaan/minimal-text-diffusion#3]
Hi,

Thank you very much for this clean code. I have a question about the tT_loss at [here](https://github1s.com/madaan/minimal-text-diffusion/blob/HEAD/src/modeling/diffusion/gaussian_diffusion.py#L289-L292). 

According to equation (2) in the diffusion-lm paper, I could not find this loss. As in your README, this loss is used for making sure mean=0 for x_T. However, wouldn't it be by default with the diffusion process? And in the original DDPM paper, I also couldn't find this loss.

**Comment by madaan** [comment]
Great question!

The formulation is more closely reflected in [this blog](https://lilianweng.github.io/posts/2021-07-11-diffusion-models/).

Specifically, the loss can be broken down into three parts:


![image](https://user-images.githubusercontent.com/1304693/202792981-2adc4d92-c6cf-4b42-a108-1344aadb0339.png)


The `tT_loss` corresponds to $L_T$. 

$L_T$ can typically be ignored as it is a constant, because $x_0$ is not a parameter(perhaps why it's not included in the original implementation). In the training loop, only one of $L_{t-1}$ [is used](https://github.com/madaan/minimal-text-diffusion/blob/cdda07fbc534b2272afca5c65e19058dbbd77c3d/src/modeling/diffusion/gaussian_diffusion.py#L278).

However, in our case, the $x_0$ are the embeddings that we train from scratch! So this loss is included. I suspect that things will not break even if we drop it, as there will be points in training when $t=T$.

Hope this makes sense. I'll be happy to add more details if needed.

**Comment by BaohaoLiao** [comment]
Makes sense to me now. Thank you.
